### ISSUE TITLE
"what-input" library doesn't work correctly when pressing key for first time

### ISSUE THREAD

**Issue by sophieH29** [microsoft/fluent-ui-react#434]
### Steps

1. Have a menu with three items,
2. click on the third item,
3. hit left arrow,
4. hit left arrow again.

### Expected Result
- The second item selected and highlighted after step 3.
- The first item selected and highlighted after step 4.

### Actual Result
After step 3, the second item is correctly selected, but it is not highlighted 💣

![menu_keyboard](https://user-images.githubusercontent.com/9615899/46717794-6e5cad80-cc69-11e8-8651-e07ae9ab4a09.gif)

### Version
Current master

### The root cause
Scenario:
- ```onKeyDown``` handler is called on a ```FocusZone```
- inside this handler, there is a quite big logic, which at the end finds a particular child ```MenuItem``` and focuses it
- ```onFocus``` handler of ```MenuItem``` is called where it is checked if the ```whatInput.ask() === 'keyboard'```, we will get here ```false```
- and only afterward the ```window.addEventListener('keydown')``` from ```what-input``` lib is called and update a state to ```keyboard``` but it's too late.

#### With styles
```what-input``` library adds attribute ```data-whatinput``` to html with updated value of what was the last input (for e.g. keyboard, mouse, etc)
So in styles we use it as
```js
[data-whatintent='mouse'] *:focus {
  outline: none;
}
```

**Pros:**
- it we solve all possible issues with handling it with scripts and we'll get rid of re-rendering component when we focus ```MenuItem```

**Cons:**
- Currently Fela doesn't allow such global selectors used in particular component

#### Update state of ```what-input``` manually

Update what-input state by just calling its function, so in FocusZone it will look like:
```js
 private _onKeyDown = (ev: React.KeyboardEvent<HTMLElement>): boolean | undefined => {
    ...
    // call it beforehand and update state from 'mouse' to 'keyboard'
    whatInput.setInput(ev)
   ...
```

**Pros:**
- Full and transperent control of state

**Cons:**
- ```what-input``` currently does not expose function for triggering a state change manually. I'll create an issue in their repo, but we can't be assure they'll approve. Meanwhile, we can add this library to Stardust, as there is no a lot of code and also remove parts that we're not using.

**Comment by bmdalex** [comment]
duplicate of #344

**Comment by sophieH29** [comment]
The library author might fix it by extending library API https://github.com/ten1seven/what-input/issues/80
Waiting for their response.
